jaxley_mech/synapses/ribbon.py

```diff
@@ -17,62 +16,81 @@ def __init__(self, name: Optional[str] = None):
-            f"{name}_docked": 4,  # Number of vesicles at the dock
-            f"{name}_ribboned": 25,  # Number of vesicles at the ribbon
-            f"{name}_P_rel": 0,  # Normalized vesicle release
-            f"{name}_P_s": 0,  # Kernel of postsynaptic conductance
+            f"{name}_exo": 0.75,  # Number of vesicles released
+            f"{name}_RRP": 1.5,  # Number of vesicles at the dock
+            f"{name}_IP": 5.0,  # Number of vesicles at the ribbon
+            f"{name}_RP": 12.5,  # Number of vesicles in the reserve pool
         }
         self.META = META
 
     def update_states(self, u, delta_t, pre_voltage, post_voltage, params):
-        """Return updated synapse state."""
+        """
+        Return updated synapse state.
+        
+        Caution: Synaptic states currently solved with explicit Euler!
+        """
         name = self.name
-        k = 1.0
-        V_half = -35
 
         # Presynaptic voltage to calcium to release probability
-        p_d_t = 1 / (1 + save_exp(-k * (pre_voltage - V_half)))
-
-        # Vesicle release (NOTE: p_d_t is the mean of the beta distribution)
-        new_released = p_d_t * u[f"{name}_docked"]
-
-        # Movement to the dock
-        new_docked = (
-            u[f"{name}_docked"]
-            + params[f"{name}_p_r"] * u[f"{name}_ribboned"]
-            - new_released
+        p_d_t = 1.0 / (
+            1.0
+            + save_exp(
+                -1 * params[f"{name}_k"] * (pre_voltage - params[f"{name}_V_half"])
+            )
         )
-        new_docked = jnp.clip(new_docked, 0, params[f"{name}_D_max"])
 
-        # Movement to the ribbon
-        dock_moved = jnp.maximum(0, new_docked - u[f"{name}_docked"])
-        new_ribboned = u[f"{name}_ribboned"] + params[f"{name}_lam"] - dock_moved
-        new_ribboned = jnp.clip(new_ribboned, 0, params[f"{name}_R_max"])
+        # Glutamate release
+        e_t = (
+            params[f"{name}_e_max"]
+            * p_d_t
+            * u[f"{name}_RRP"]
+            / params[f"{name}_RRP_max"]
+        )
+        # Rate of RP --> IP, movement to the ribbon
+        r_t = (
+            params[f"{name}_r_max"]
+            * (1 - u[f"{name}_IP"] / params[f"{name}_IP_max"])
+            * u[f"{name}_RP"]
+            / params[f"{name}_RP_max"]
+        )
+        # Rate of IP --> RRP, movement to the dock
+        i_t = (
+            params[f"{name}_i_max"]
+            * (1 - u[f"{name}_RRP"] / params[f"{name}_RRP_max"])
+            * u[f"{name}_IP"]
+            / params[f"{name}_IP_max"]
+        )
+        # Rate of RP refilling
+        d_t = params[f"{name}_d_max"] * u[f"{name}_exo"]
 
-        # Single exponential decay to model postsynaptic conductance (Dayan & Abbott)
-        P_rel = new_released / params[f"{name}_D_max"]
-        P_s = save_exp(-delta_t / params[f"{name}_tau"])
+        # Calculate the new vesicle numbers
+        new_RP = u[f"{name}_RP"] + (d_t - e_t) * delta_t
```

Comment:
hmmm, notice a bug, this should be

```
new_RP = u[f"{name}_RP"] + (d_t - r_t) * delta_t
```

but I will fix it in the next "solvers" PR.